exec.js

```diff
@@ -12,29 +12,106 @@ var run = function() {
+
+    for(var i = 0; i < containers.length; i++) {
+        // activation refers to a checkbox that signifies if the user has activated
+        // the argument (only for optional arguments)
+        var activation = containers[i].getElementsByClassName("activation")[0];
+        if (activation && activation.checked || !activation) {
+            var option = containers[i].getElementsByTagName("input");
+            // to handle dialog boxes
+            if (option.length == 1 && activation) {
+                var option = containers[i].getElementsByTagName("select")[0];
+            }
+            // to handle radio buttons
+            else if (option.length > 2) {
+                for(var k = 0; k < option.length; k++) {
+                    if (option[k].checked && k != 0) {
+                        var option = option[k]
+                        break;
+                    }
+                }
+            }
+            else
+            // for all other inputs, relevant input is always the last one
+                option = option[option.length - 1];
+
+            var optionName = option.name.replace("arg-", "");
+            optionName = optionName.replace("-activated", "");
+            var optionValue;
+            
+            // if the arg has a flag, prepend the flag
+            var flagArray = state.run.get("plugins[" + selectedIndex + "].args");
+            var flagIndex;
+            for(var z = 0; z < flagArray.length; z++) {
+                if (flagArray[z].name == optionName) {
+                    flagIndex = z;
+                    break;
+                }
+            }
+            var flag = flagArray[flagIndex].flag;
+
+            if (option.value == "on" && option.type == "checkbox") {
+                optionValue = flag? flag + " " + option.value.replace("on", "") : null;
+            } else 
+                optionValue = flag ? flag + " " + option.value : option.value;
+
+            args.push([optionName, optionValue]);
+        }
+    }
+
+    var command = state.run.get("plugins[" + selectedIndex + "].command");
+
+    for(var x = 0; x < args.length; x++) {
+        // args is an array of (flag name, flag value) tuples
+        command = command.replace("$" + args[x][0], args[x][1]);
+    }
+
+    // gets rid of any variables that have not been specified
+    command = command.replace(new RegExp(/\$[a-z][a-z0-9_]*\s/g), "");
+
+    dialog.showMessageBox({"type": "info", "message": command, "buttons": ["close"]});
+};
+
+/* Toggles an overlay display with arguments for the user
+ * to enter */
 var toggleOverlay = function() {
     el = document.getElementById("overlay");
     el.style.visibility = (el.style.visibility == "visible") ? "hidden" : "visible";
     window.scrollTo(0,0);
 };
+
+var formValid = function() {
+    var argForm = document.getElementById("arg-form");
+    for(var i = 0; i < argForm.length; i++) {
+        if (!argForm[i].validity.valid)
+            return false;
```

Comment:
Just to understand, here `.valid` is true if either there is no validating regular expression or there is a regular expression that matches, correct?